pkg/authentication/handlers.go

```diff
@@ -123,6 +169,37 @@ func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
 	}
 
 	_ = json.NewEncoder(w).Encode(tokens)
+
+}
+
+func (a *API) nonceCheck(r *http.Request, idToken *Principal) error {
+	nonce := a.cookieManager.GetNonceCookie(r)
+	if nonce == "" {
+		a.logger.Error(nonceCookieNotFoundMsg)
+		return nonceCookieNotFoundErr
+	}
+
+	if idToken.Nonce != nonce {
+		a.logger.Error(idTokenNonceMatchMsg)
+		return idTokenNonceMatchErr
+	}
+
+	return nil
+}
+
+func (a *API) stateCheck(r *http.Request, state string) error {
+	stateCookieValue := a.cookieManager.GetStateCookie(r)
+	if stateCookieValue == "" {
+		a.logger.Error(stateCookieNotFoundMsg)
+		return stateCookieNotFoundErr
+	}
+
+	if stateCookieValue != state {
+		a.logger.Error(stateParamMatchMsg)
+		return stateParamMatchErr
+	}
+
+	return nil
```

Comment:
i wonder if this should be deemed "business logic" and moved to a `service.go`

Reply:
I don't agree because this being "request validation" it belongs to the controller layer.
The only "out of tone" parte is checking the IDToken "nonce" claim but it wouldn't be worth it to separate that if check